node-mixin/dashboards/use.libsonnet

```diff
@@ -0,0 +1,151 @@
+local g = import 'grafana-builder/grafana.libsonnet';
+
+{
+  grafanaDashboards+:: {
+    'node-cluster-rsrc-use.json':
+      local legendLink = '%s/dashboard/file/k8s-node-rsrc-use.json' % $._config.grafana_prefix;
+
+      g.dashboard('USE Method / Cluster')
+      .addRow(
+        g.row('CPU')
+        .addPanel(
+          g.panel('CPU Utilisation') +
+          g.queryPanel('instance:node_cpu_utilisation:avg1m * instance:node_num_cpu:sum / scalar(sum(instance:node_num_cpu:sum))', '{{instance}}', legendLink) +
+          g.stack +
+          { yaxes: g.yaxes({ format: 'percentunit', max: 1 }) },
+        )
+        .addPanel(
+          g.panel('CPU Saturation (Load1)') +
+          g.queryPanel(|||
+            instance:node_cpu_saturation_load1: / scalar(sum(up{%(nodeExporterSelector)s}))
+          ||| % $._config, '{{instance}}', legendLink) +
+          g.stack +
+          { yaxes: g.yaxes({ format: 'percentunit', max: 1 }) },
+        )
+      )
+      .addRow(
+        g.row('Memory')
+        .addPanel(
+          g.panel('Memory Utilisation') +
+          g.queryPanel('instance:node_memory_utilisation:ratio', '{{instance}}', legendLink) +
```

Comment:
Ratio of what?

Reply:
Strictly, it's `MemAvailable_per_MemTotal`, but then one minus that, so could be expressed as `MemUsed_per_MemTotal` but `MemUsed` doesn't exist in an actual metric name, so we could as well say `used_per_total`, resulting in the name `instance:node_memory_used_per_total:ratio`. However, `utilisation` is essentially the same as `used_per_total` just easier to the tongue. Given that we cannot directly use the original metric names (due to the `1 - ...`), I think `instance:node_memory_utilisation:ratio` is the least bad name in this situation.